NMS-14212: Event docs cleanup (#5002)
OpenNMS/opennms: docs/modules/operation/pages/events/event-configuration.adoc

== The eventd-configuration.xml file

The overall behavior of Eventd is configured in the `$\{OPENNMS_HOME}/etc/eventd-configuration.xml` file.
This file does not need to be changed in most installations.
The configurable items include the following:

TCPAddress::
    The IP address to which the Eventd XML/TCP listener will bind.
    Defaults to `127.0.0.1`.
TCPPort::
    The TCP port number on `TCPAddress` to which the Eventd XML/TCP listener will bind.
    Defaults to `5817`.
UDPAddress::
    The IP address to which the Eventd XML/UDP listener will bind.
    Defaults to `127.0.0.1`.
UDPPort::
    The UDP port number on `TCPAddress` to which the Eventd XML/UDP listener will bind.
    Defaults to `5817`.
receivers::
    The number of threads allocated to service the event intake work done by Eventd.
queueLength::
    The maximum number of events that may be queued for processing.
    Additional events will be dropped.
    Defaults to unlimited.
getNextEventID::
    An SQL query statement used to retrieve the ID of the next new event.

>     Changing this setting is not recommended.

Reviewer: It would depend on how severe the consequences would be. I'd like to minimize the use of admonitions. @mmahacek do you know how serious it is if they did change the setting?

>     Changing this setting is not recommended.

Author: I don't recall ever changing this setting.

socketSoTimeoutRequired::
    Whether to set a timeout value on the Eventd receiver socket.
socketSoTimeoutPeriod::
    The socket timeout, in milliseconds, to set if `socketSoTimeoutRequired` is set to `yes`.
logEventSummaries::
    Whether to log a simple (terse) summary of every event at level `INFO`.
    Useful when troubleshooting event processing on busy systems where `DEBUG` logging is not practical.

The set of known events is configured in `$\{OPENNMS_HOME}/etc/eventconf.xml`.

> This file opens with a `<global>` element, whose `<security>` child element defines which event fields may not be overridden in the body of an event submitted via any Eventd listener.

Author: The action being done is reading from the file, which would be an "override" as opposed to saving change and writing over an existing value.  But it could probably go either way.  I don't have strong feelings on this and will go with whichever you think is best here.

> This file opens with a `<global>` element, whose `<security>` child element defines which event fields may not be overridden in the body of an event submitted via any Eventd listener.

Reviewer: Thanks, I wasn't sure if there was a distinction, so let's keep override.

> This file opens with a `<global>` element, whose `<security>` child element defines which event fields may not be overridden in the body of an event submitted via any Eventd listener.

Author: I dunno.  That's just based on the logic in my head at the moment. 😬

----

Every event definition has this same basic structure.
See <<events/event-definition.adoc#ga-events-anatomy-of-an-event, Anatomy of an Event>> for a discussion of the structural elements.

[[severities]]

> === A word about severities

Reviewer 2: ```suggestion
=== A word about severities

```

When setting event severities, it's important to consider each event in the context of your infrastructure as a whole.
Events whose severity is critical at the zoomed-in level of a single device may not merit a `Critical` severity in the zoomed-out view of your entire enterprise.

> Since an event with `Critical` severity can never have its alarms escalated, you should usually reserve this highest severity level for events that unequivocally indicate a truly critical impact to the business.

Reviewer: Please keep "highest" in. I added it to make it apparent why the severity level cannot be escalated.

Rock legend Nigel Tufnel offered https://www.youtube.com/watch?v=4xgx4k83zzc[some wisdom] on the subject.

> === Structure of the `eventconf.xml` tributary files

Reviewer 2: ```suggestion
=== Structure of the `eventconf.xml` tributary files

```

----

Every event definition has this same basic structure.

> See <<events/event-definition.adoc, Anatomy of an Event>> for a discussion of the structural elements.

Author: it's building for me locally.  will check on circleci results after my next push.

----

Every event definition has this same basic structure.
See <<events/event-definition.adoc#ga-events-anatomy-of-an-event, Anatomy of an Event>> for a discussion of the structural elements.

[[severities]]
=== A word about severities
When setting event severities, it's important to consider each event in the context of your infrastructure as a whole.
Events whose severity is critical at the zoomed-in level of a single device may not merit a `Critical` severity in the zoomed-out view of your entire enterprise.
Since an event with `Critical` severity can never have its alarms escalated, you should usually reserve this highest severity level for events that unequivocally indicate a truly critical impact to the business.
Rock legend Nigel Tufnel offered https://www.youtube.com/watch?v=4xgx4k83zzc[some wisdom] on the subject.

=== The structure of the `eventconf.xml` tributary files
The order of event definitions is very important, as an incoming event is matched against them in order.

> It is possible and often useful to have several event definitions that could match variant forms of a given event; for example, based on the values of SNMP trap variable bindings.

Reviewer 2: ```suggestion
It is possible, and often useful, to have several event definitions that could match variant forms of a given event; for example, based on the values of SNMP trap variable bindings.
```

> * To save memory and shorten startup times, you may wish to remove event definition files that you know you do not need.
> * If you need to customize some events in one of the default tributary files, you may wish to make a copy of the file containing only the customized events, and slot the copy above the original; this practice will make it easier to maintain your customizations in case the default file changes in a future release of {page-component-title}.

Author: ```suggestion

* To save memory and shorten startup times, you may want to remove event definition files that you know you do not need.
* If you need to customize some events in one of the default tributary files, you may want to make a copy of the file containing only the customized events and load the copy above the original in `eventconf.xml`.
This practice will make it easier to maintain your customizations in case the default file changes in a future release of {page-component-title}.
```

> * If you need to customize some events in one of the default tributary files, you may wish to make a copy of the file containing only the customized events, and slot the copy above the original; this practice will make it easier to maintain your customizations in case the default file changes in a future release of {page-component-title}.

Author: @emarsh1995 What do you think about this tweak of the text?

> * If you need to customize some events in one of the default tributary files, you may wish to make a copy of the file containing only the customized events, and slot the copy above the original; this practice will make it easier to maintain your customizations in case the default file changes in a future release of {page-component-title}.

Reviewer 2: I like it! Breaking the second bullet point into two sentences is a good idea. :)
